README.md

```diff
@@ -1,3 +1,88 @@
 # ioc-hierarchy-service
 In-memory queryable hierarchy service
 
+This project generates a debian package with all components necessary to
+create an in-memory cache of a hierarchy. The hierarchy is modelled as
+a graph and can be queried in a number of ways. The cache itself is run
+as a daemon process which can be backed up by using another daemon
+process called backup service. All interprocess communication is handled
+with ZeroMQ.
+
+Querying the process must be done using the bundled python module. This
+can be imported and a function named `query` can be run, e.g. to check
+the health of the service:
+
+```
+>>> import tag_hierarchy_client
+>>> tag_hierarchy_client.query([{'command': 'healthcheck'}], "server")
+```
+
+# Architectural overview
+The following diagram describes which components are bundled with the
+debian package and how they are intended to be used:
+
+![Architecture diagram][docs/hierarchy_daemon.png]
+
+## Communication
+Components communicate via ZeroMQ(ZMQ). We use the patterns REQ-REP
+(request-response) when the python client communicates with the server
+or the statemanager.
+
+From the python perspective, a request has to pass a list of dictionaries
```

Comment:
"perspective" is maybe not such a good word.

Reply:
Ok, it is what I mean though. What would be better, point of view?

Reply:
I think you mean that "the user of the python client will pass a command request as a dictionary."